bin/lib/getOptions.js

```diff
@@ -41,7 +42,7 @@ export default async function getOptions({ argv, env, flags, log, packageJson })
     exitZeroOnChanges: flags.exitZeroOnChanges === '' ? true : flags.exitZeroOnChanges,
     exitOnceUploaded: flags.exitOnceUploaded === '' ? true : flags.exitOnceUploaded,
     ignoreLastBuildOnBranch: flags.ignoreLastBuildOnBranch,
-    preserveMissingSpecs: flags.preserveMissing || !!flags.only,
+    preserveMissingSpecs: flags.preserveMissing || !!flags.only || flags.onlyChanged !== undefined,
```

Comment:
This is now done automatically by the backend, `preserveMissingSpecs` actually has a slightly different effect (e.g. you can't remove stories any more!).